src/dev/lupio/lupio_rtc.cc

```diff
@@ -0,0 +1,103 @@
+#include "dev/lupio/lupio_rtc.hh"
+
+#include <ctime>
+#include <string>
+
+#include "base/bitfield.hh"
+#include "base/time.hh"
+#include "base/trace.hh"
+#include "debug/LupioRTC.hh"
+#include "mem/packet.hh"
+#include "mem/packet_access.hh"
+#include "params/LupioRTC.hh"
+#include "sim/system.hh"
+
+namespace gem5
+{
+
+using namespace RiscvISA;
+
+LupioRTC::LupioRTC(const Params &params) :
+    BasicPioDevice(params, params.pio_size),
+    system(params.system),
+    time(params.time)
+{
+    struct timeval tv;
+    struct tm time_bd;
+
+    gettimeofday(&tv, NULL);
+    gmtime_r(&tv.tv_sec, &time_bd);
+
+    DPRINTFN("Real-time clock set to %s", asctime(&time_bd));
+}
+
+LupioRTC::~LupioRTC()
```

Comment:
Don't need this function if it's empty. If it's pure virtual (which I don't think it is...) I *think* you can use `= default` in the header.